pkg/agent/agentserver/server.go

```diff
@@ -201,7 +273,7 @@ func (s *ProxyServer) serveRecvFrontend(stream agent.ProxyService_ProxyServer, r
 			// the address, then we can send the Dial_REQ to the
 			// same agent. That way we save the agent from creating
 			// a new connection to the address.
-			backend, err = s.randomBackend()
+			backend, err = s.BackendGetter.Backend()
```

Comment:
Maybe `getSingleBackend`? Just something to indicate that we're picking one backend out of multiple.

Reply:
+1

Reply:
I made the BackendManager interface, which has `Backend()` and `AllBackends()` methods, let me know if their meaning is still not obvious.

Reply:
That looks good. Thanks!